tests/test_api.py

```diff
@@ -1284,7 +1284,7 @@ def test_test_endpoints(self):
 
         assert self.app.post("/tests/events/drop").status_code == 200
         dataset = get_dataset("events")
-        writer = dataset.get_table_writer()
+        writer = dataset.get_writable_storage().get_table_writer()
```

Comment:
This isn't going to type check. You can fix it like this:

```suggestion
        storage = dataset.get_writable_storage()
        assert storage is not None
        writer = storage.get_table_writer()
```

It'd be a non-functional change since the test obviously fail if there is no writable storage anyway, but it seems like a good idea to avoid the noise if we'd like to eventually tighten up type constraints in tests too at some point.

Reply:
Forgot about these since I wasn't running with strict